### ISSUE TITLE
RFC: Friends' progress

### ISSUE THREAD

**Issue by Birdulon** [z411/trackma#171]
Keeping tabs on the show progress of other people can be handy for discussion purposes, avoiding inadvertent spoilers etc. Sure you can always ask them, but it would be even more convenient to see at a glance what all the safe seasonal discussions are for each friend.
### Main Proposal

The addition of an (optional) element to shows (or manga/other tracked items) that displays the progress of one or several selected other users on that API, as well as special highlights indicating relative progress - i.e. whether they are behind, caught up with, or ahead of you in the series. My main target is the Qt client, however much of this could also be implemented in the Gtk or Curses clients.
#### Initial Suggestions
1. A compact, information-dense approach - additional subbars drawn above the episode subbar on the progressbar widget.
   - This may not be very intuitive, and would likely need a legend somewhere in the UI to remind the user who each subbar corresponds to.
2. A widescreen approach - additional progress string columns for each selected user.
   - This column need not be a full progress string, it could just be the number of episodes watched, or even an icon showing the relative progress.
3. A more scalable approach - a column which contains short aliases of each selected user watching the show. The aliases would be ordered and coloured according to their relative progress.
#### Main Considerations
- Visual Consistency
  - The indicators should not clash with the existing Trackma GUI
  - Indicators should match the system theme as much as practical
- Readability
  - Key differences should be quickly discernable in large lists
- Scalability
  - Up to ten or so users should be handled gracefully
  - Perhaps a more minimal indication method could be used for more users
- Data Handling
  - Other user's lists should be stored in a way that is cohesive with Trackma's existing data structures
### Closing

I'm fully prepared to to implement the above 3 suggestions for libmal. Before I undergo this, I'd like some feedback regarding the ideas and would welcome any suggestions for alternate means of indication. While I'm most interested in what @z411 might have to say, anything from Trackma contributors or users would be much appreciated.
Thanks for reading!

**Comment by archnous** [comment]
I'm not very social either; I would prefer multiple library directories instead.

**Comment by hbers** [comment]
@archnous This. This makes more sense and an addition to that a basic rss reader/torrent downloader would be great if you have the time to add new features.

**Comment by z411** [comment]
@archnous This is something that was proposed time ago, it's relatively trivial to implement but can't symbolic links work to the same end?

@hbers I worked on the torrent functionality last year (`torrent` branch) but I never worked on the interface part of it. But I plan to.

**Comment by ToostInc** [comment]
While on the one hand this does seem rather complex it *would* be a feature no other similar anime tracker offers, as far as I know. Not sure how much app-exclusive features matter to people though, especially in an open source ecosystem. 

Although I agree with @z411 that the basic list managing should stay the focus, I'd you're willing to work on it I think it's probably worth a shot. 

Don't know if adding it in with your own progress would be the best option though, I can see that getting busy really quickly one you add more than two friends. Maybe under a tab of its own and then list (shared) show progress per user?

If you manage to get it working with MAL I can try to port it for Anilist as well...someday at least.

**Comment by archnous** [comment]
@z411 Yes, symbolic links is my workaround ATM. Another thing I've noticed is that a directory with too many subdirs/files breaks the Tracker:
```
Tracker: inotify: Watching /storage/subdir1/subdir2/...
Tracker: Watch directory not found! Tracker will stop.
```

**Comment by hbers** [comment]
@z411 Glad to hear that. Seriously the torrent functionality would make this client perfect.

**Comment by z411** [comment]
@hbers See #176

**Comment by Birdulon** [comment]
For anyone who's interested, I've put up basic functionality over at [Friends branch in Birdulon/trackma](/Birdulon/trackma/tree/Friends). Currently it only supports one representation (text with usernames shortened to 3 characters) and only with MyAnimeList but I plan to explore other representations and adding support in other APIs is trivial (I have no friends using other sites at this time but I'd be more than happy to add support with other APIs on request).
![trackmafriends](https://cloud.githubusercontent.com/assets/5397662/11654671/875adc24-9df7-11e5-82c7-73f8a61a7f88.png)
![trackmafriends1](https://cloud.githubusercontent.com/assets/5397662/11654706/ae1a5ccc-9df7-11e5-8c7f-aac2be7e038a.png)

**Comment by wizbright** [comment]
I would enjoy this function for Anilist. Seems really neat :+1:

**Comment by Birdulon** [comment]
Added anilist support. (321df2063)

**Comment by FichteFoll** [comment]
I only now saw this issue. Many years ago, I've written a small web application that utilizes the AniList API to provide similar functionality for tracking the progress among shared anime of a configurable list of users. You may use it here: https://fichtefoll.github.io/anitogether.

Personally, I don't think this is functionality that should be embedded within trackma as it seems quite niche and does not require a desktop-level application, as shown above.

**Comment by Birdulon** [comment]
At the risk of sounding too snarky, nothing trackma does outside of
incrementing the episode progress number when you use a desktop media
player requires a desktop-level application. When I encountered python
dependency issues after a Manjaro update I didn't even bother fixing it, I
just went back to using the Anilist website.
